shared/js/background/classes/tab.es6.js

```diff
@@ -214,47 +201,30 @@ class Tab {
         return policy.resourcePermitted(resourcePath)
     }
 
-    resetBadgeIcon () {
-        // set the new tab icon to the dax logo
-        browserWrapper.setBadgeIcon({ path: '/img/icon_48.png', tabId: this.id })
-    }
-
-    updateBadgeIcon (target) {
-        if (this.site.specialDomainName) return
-        let gradeIcon
-        const grade = this.site.grade.get()
-
-        if (this.site.isContentBlockingEnabled()) {
-            gradeIcon = gradeIconLocations[grade.enhanced.grade]
-        } else {
-            gradeIcon = gradeIconLocations[grade.site.grade]
-        }
-
-        const badgeData = { path: gradeIcon, tabId: this.id }
-        if (target) badgeData.target = target
-
-        browserWrapper.setBadgeIcon(badgeData)
-    }
-
     updateSite (url) {
         if (this.site.url === url) return
 
         this.url = url
         this.site = new Site(url, this._tabState)
-
-        // reset badge to dax whenever we go to a new site
-        this.resetBadgeIcon()
     }
 
     // Store all trackers for a given tab even if we don't block them.
-    addToTrackers (t) {
+    /**
+     * @param t
+     * @param {string} baseDomain
+     * @param {string} url
+     * @returns {Tracker}
+     */
+    addToTrackers (t, baseDomain, url) {
         const trackers = this.trackers
         const tracker = this.trackers[t.tracker.owner.name]
 
         if (tracker) {
-            tracker.addTrackerUrl(t)
+            tracker.addTrackerUrl(t, this.url || '', baseDomain, url)
```

Comment:
I think we could probably clean this up to not need the `t.tracker.owner.name` keying here but let's leave that to a follow up.